### PR TITLE
Feat/algorand chain

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
I am opening this PR so that the Algorand chain code can be peer-reviewed and QA-tested. This code will be part of a larger PR, including the Tinyman connector.

Other than implementing the Algorand chain, this PR also attempts to centralize the places where the chains are being built.

**Tests performed by the developer**:

I have tested all new endpoints.

**Tips for QA testing**:

You can use Pera wallet for Algorand. We already have a funded internal wallet for testing.
Please test the following endpoints for Algorand:
- `/network/status` should return active algorand instances
- `/wallet`, `/wallet/add`, `/wallet/remove` should work as expected for Algorand (`/wallet/sign` not implemented)
- all of the `/algorand/...` endpoints

Please also test creating all other chains as some of the changes in this PR affect chain instantiation. You can do that by simply requesting chain tokens for each chain.

[ch-39262]

### CODE REVIEW

**Comment by rxlxrxsx** [comment]
Test run: https://hummingbot.testrail.io/index.php?/milestones/view/64

File: test-helpers/curl/requests/algorand_balances.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "chain": "algorand",
+  "network": "mainnet",
+  "address": "FJZ4AJ3EWSNV4PXULFTTT4R5PLMNASEIMWEK5H4EY6E67RGJNSEY7OZEMA",
```
Comment (vic-en) [inline test-helpers/curl/requests/algorand_balances.json:4]:
```suggestion
  "address": "$ALGORAND_ADDRESS",
```

File: test-helpers/curl/requests/algorand_opt_in.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "network": "testnet",
+  "address": "Y6E7QWFTAI4KWJ4DNSJOLDZ74VU454X5XZLBEPEWMFI7V4GSJ72X3O27HQ",
```
Comment (vic-en) [inline test-helpers/curl/requests/algorand_opt_in.json:3]:
```suggestion
  "address": "$ALGORAND_ADDRESS",
```

File: test-helpers/curl/requests/remove_algorand_key.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "address": "FJZ4AJ3EWSNV4PXULFTTT4R5PLMNASEIMWEK5H4EY6E67RGJNSEY7OZEMA",
```
Comment (vic-en) [inline test-helpers/curl/requests/remove_algorand_key.json:2]:
```suggestion
  "address": "$ALGORAND_ADDRESS",
```

**Review by vic-en — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
     "@injectivelabs/wallet-ts": "^1.0.300",
     "@pancakeswap/sdk": "^2.4.5",
     "@pangolindex/sdk": "^1.1.0",
-    "@perp/sdk-curie": "^1.16.0",
+    "@perp/sdk-curie": "^1.18.0",
```
[the remark this answers is not in the document]
Comment (petioptrv) [reply package.json:43]: It's used by the algorand SDK

File: src/chains/algorand/algorand.ts [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+  public get storedAssetList(): AlgorandAsset[] {
+    return Object.values(this._assetMap);
+  }
+
+  public ready(): boolean {
+    return this._ready;
+  }
+
+  public async init(): Promise<void> {
+    // todo: common EVM-like interface
```
[the remark this answers is not in the document]
Comment (petioptrv) [reply src/chains/algorand/algorand.ts:73]: done

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (vic-en) [reply package.json:43]: That can't be the case. Pls double check.

**Review by rxlxrxsx — APPROVED** [review]
Tested and confirmed working by QA.

- Successfully cloned and installed branch
- Setup and run gateway, no issues found
- Export needed variables and run endpoints for algorand and confirmed working
  - Network status and all enpoints for algorand
  - Wallets (except for sign) for algorand
  - Requesting chain tokens for each chain

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (petioptrv) [reply package.json]: You're right, reverted

**Comment by vic-en** [comment]
@petioptrv I think commit [3cd2773](https://github.com/CoinAlpha/gateway/pull/15/commits/3cd277326f8bde399b1fed278e51e7a82afdab36) doesn't belong in this pr

**Comment by vic-en** [comment]
Corresponding pr on foundation - https://github.com/hummingbot/gateway/pull/90
PRP - https://snapshot.org/#/hbot-prp.eth/proposal/0x08a53dead66043ee8e8b0e65229a99b52be2e212a640e992eb72c9b4cd388701

**Pull request closed without merging**